sklearn/linear_model/sag_fast.pyx

```diff
@@ -0,0 +1,325 @@
+# cython: cdivision=True
+#
+# Authors: Danny Sullivan <dbsullivan23@gmail.com>
+#
+# Licence: BSD 3 clause
+
+import numpy as np
+cimport numpy as np
+import warnings
+from libc.math cimport fabs, exp, log
+from libc.time cimport time, time_t
+
+from ..utils.seq_dataset cimport SequentialDataset
+from .sgd_fast cimport Log, SquaredLoss, LossFunction
+
+cdef extern from "sgd_fast_helpers.h":
+    bint skl_isfinite(double) nogil
+
+# This sparse implementation is taken from section 4.3 of "Minimizing Finite
+# Sums with the Stochastic Average Gradient" by
+# Mark Schmidt, Nicolas Le Roux, Francis Bach. 2013. <hal-00860051>
+#
+# https://hal.inria.fr/hal-00860051/PDF/sag_journal.pdf
+
+
+cdef inline double fmax(double x, double y) nogil:
+    if x > y:
+        return x
+    return y
+
+
+def sag_sparse(SequentialDataset dataset,
```

Comment:
This is an artifact of me implementing both standard and sparse implementations in the pyx file. Because I removed the standard implementation, it's fine to remove the ```_sparse``` implementation prefix.